src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProject.cs

```diff
@@ -358,7 +358,7 @@ private void OnBatchScopeDisposed()
                         solutionChanges,
                         documentFileNamesAdded,
                         documentsToOpen,
-                        (s, documents) => solution.AddDocuments(documents),
+                        (s, documents) => s.AddDocuments(documents),
```

Comment:
With C# 8 this can be:

```suggestion
                        (solution, documents) => solution.AddDocuments(documents),
```